inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/mongodb-cdc/src/main/java/org/apache/inlong/sort/mongodb/MongoDBTableSource.java

```diff
@@ -0,0 +1,331 @@
+            Optional.ofNullable(copyExisting).ifPresent(builder::copyExisting);
+            Optional.ofNullable(copyExistingQueueSize).ifPresent(builder::copyExistingQueueSize);
+            Optional.ofNullable(batchSize).ifPresent(builder::batchSize);
+            Optional.ofNullable(pollMaxBatchSize).ifPresent(builder::pollMaxBatchSize);
+            Optional.ofNullable(pollAwaitTimeMillis).ifPresent(builder::pollAwaitTimeMillis);
+            Optional.ofNullable(heartbeatIntervalMillis)
+                    .ifPresent(builder::heartbeatIntervalMillis);
+
+            return SourceFunctionProvider.of(builder.build(), false);
+        }
+    }
+
+    protected MetadataConverter[] getMetadataConverters() {
+        if (metadataKeys.isEmpty()) {
+            return new MetadataConverter[0];
+        }
+
+        return metadataKeys.stream()
+                .map(
+                        key ->
+                                Stream.of(MongoDBReadableMetadata.values())
+                                        .filter(m -> m.getKey().equals(key))
+                                        .findFirst()
+                                        .orElseThrow(IllegalStateException::new))
+                .map(MongoDBReadableMetadata::getConverter)
+                .toArray(MetadataConverter[]::new);
+    }
+
+    @Override
+    public Map<String, DataType> listReadableMetadata() {
+        return Stream.of(MongoDBReadableMetadata.values())
+                .collect(
+                        Collectors.toMap(
+                                MongoDBReadableMetadata::getKey,
+                                MongoDBReadableMetadata::getDataType));
+    }
+
+    @Override
+    public void applyReadableMetadata(List<String> metadataKeys, DataType producedDataType) {
+        this.metadataKeys = metadataKeys;
+        this.producedDataType = producedDataType;
+    }
+
+    @Override
+    public DynamicTableSource copy() {
+        MongoDBTableSource source =
+                new MongoDBTableSource(
+                        physicalSchema,
+                        hosts,
+                        username,
+                        password,
+                        database,
+                        collection,
+                        connectionOptions,
+                        copyExisting,
+                        copyExistingQueueSize,
+                        batchSize,
+                        pollMaxBatchSize,
+                        pollAwaitTimeMillis,
+                        heartbeatIntervalMillis,
+                        localTimeZone,
+                        enableParallelRead,
+                        splitMetaGroupSize,
+                        splitSizeMB,
+                        metricOption);
+        source.metadataKeys = metadataKeys;
+        source.producedDataType = producedDataType;
+        return source;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Should the `equals()` and `hashcode()` need to take `metricOption` into account?

Reply:
yeah ~， you are right， fixed it， thx~